packages/vm/test/t8n/t8ntool.ts

```diff
@@ -164,43 +164,78 @@ export class TransitionTool {
 
     let index = 0
 
-    this.vm.events.on('afterTx', (event) => {
-      this.afterTx(event, index, builder)
-    })
+    let trace: any[] = []
+    // Tracing
+    if (args.trace === true) {
+      this.vm.evm.events?.on('step', (e) => {
+        let hexStack = []
+        hexStack = e.stack.map((item: bigint) => {
+          return '0x' + item.toString(16)
+        })
+        const opTrace = {
+          pc: e.pc,
+          op: e.opcode.name,
+          gas: bigIntToHex(e.gasLeft),
+          gasCost: bigIntToHex(BigInt(e.opcode.fee)),
+          stack: hexStack,
+          depth: e.depth,
+          opName: e.opcode.name,
+        }
+        trace.push(JSON.stringify(opTrace))
```

Comment:
I think we should include a helper in the EVM to re-format these to the correct JSON format (or output it directly)

Reply:
Have done this with `JSONifyStepTrace`